src/main/java/duke/core/Parser.java

```diff
@@ -56,5 +57,39 @@ public Parser(String userInput) {
         }
     }
 
+    public String parseDeletePatient() throws DukeException {
+        String formattedInput;
+        String inputToParse = userInput.replace("delete patient", "").trim();
+
+        if (inputToParse.contains("#")) {
+            try {
+                formattedInput = inputToParse.replace("#", "").trim();
+                return formattedInput;
+            } catch(Exception e){
+                throw new DukeException("Please follow the format 'delete patient #<id>'.");
+            }
```

Comment:
This looks good to me. Since there is unlikely(does not make sense) to have a patient name or task description containing "#", it reduce extra parsing in DeleteCommand class